docker-bake.hcl

```
@@ -206,102 +212,208 @@ target "build-all" {
+        deps = <<EOF
+          cmake: ['3.23.4']
+          libunwind: '1.6.2'
+          mpich: ['4.0.2', '-j4']
+        EOF
       },
       {
         compiler = "clang-13"
         extra_packages = "llvm-13"
+        deps = <<EOF
+          cmake: ['3.23.4']
+          libunwind: '1.6.2'
+          mpich: ['4.0.2', '-j4']
+        EOF
       },
       {
         compiler = "clang-14"
         extra_packages = "llvm-14"
+        deps = <<EOF
+          cmake: ['3.23.4']
+          libunwind: '1.6.2'
+          mpich: ['4.0.2', '-j4']
+        EOF
       },
       {
         compiler = "clang-15"
+        deps = <<EOF
+          cmake: ['3.23.4']
+          libunwind: '1.6.2'
+          mpich: ['4.0.2', '-j4']
+        EOF
       },
       {
         compiler = "clang-16"
         distro_version = "24.04"
         extra_packages = "llvm-16"
+        deps = <<EOF
+          cmake: ['3.23.4']
+          libunwind: '1.6.2'
+          mpich: ['4.0.2', '-j4']
+        EOF
       },
       {
         compiler = "clang-16"
         distro_version = "24.04"
         extra_packages = "llvm-16 xvfb"
         variant = "vtk"
+        deps = <<EOF
+          cmake: ['3.23.4']
+          libunwind: '1.6.2'
+          mpich: ['4.0.2', '-j4']
+          mesa: ~
+          vtk: '9.3.1'
+        EOF
       },
       {
         compiler = "clang-16"
         distro_version = "24.04"
         extra_packages = "llvm-16 gfortran-13"
         fc="gfortran-13"
         variant = "zoltan"
+        deps = <<EOF
+          cmake: ['3.23.4']
+          libunwind: '1.6.2'
+          mpich: ['4.0.2', '-j4']
+          zoltan: [ -j4 ]
+        EOF
       },
       {
         compiler = "clang-17"
         distro_version = "24.04"
+        deps = <<EOF
+          cmake: ['3.23.4']
+          libunwind: '1.6.2'
+          mpich: ['4.0.2', '-j4']
+        EOF
       },
       {
         compiler = "clang-18"
         distro_version = "24.04"
+        deps = <<EOF
+          cmake: ['3.23.4']
+          libunwind: '1.6.2'
+          mpich: ['4.0.2', '-j4']
+        EOF
       },
       {
         compiler = "gcc-9"
         distro_version = "20.04"
         extra_packages = "python3-jinja2 python3-pygments"
+        deps = <<EOF
+          cmake: ['3.23.4']
+          doxygen: ['1.8.16']
+          mpich: [ '4.0.2', '-j4' ]
+        EOF
       },
       {
         compiler = "gcc-10"
         distro_version = "20.04"
+        deps = <<EOF
+          cmake: ['3.23.4']
+          mpich: ['4.0.2', '-j4']
+        EOF
       },
       {
         compiler = "gcc-10"
         distro_version = "20.04"
         extra_packages = "ssh"
         mpi_extra_flags = "--allow-run-as-root --oversubscribe"
         variant = "openmpi"
+        deps = <<EOF
+          cmake: ['3.23.4']
+          openmpi: ['v4.0', '4.0.4', '-j4']
+        EOF
       },
       {
         compiler = "gcc-11"
+        deps = <<EOF
+          cmake: ['3.23.4']
+          mpich: { args: [ '4.0.2', '-j4' ] }
```

```suggestion
          mpich: ['4.0.2', '-j4']
```

This seems to work for other builds (e.g. `gcc-10`).